auth-foundation/src/main/java/com/okta/authfoundation/credential/events/RateLimitExceededEvent.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2022-Present Okta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.okta.authfoundation.credential.events
+
+import com.okta.authfoundation.events.EventHandler
+import okhttp3.Request
+import okhttp3.Response
+
+/**
+ * Emitted via [EventHandler.onEvent] when a network request receives 429 http status code (rate limit exceeded).
+ *
+ * The default implementation retries the request 3 times, with a minimum delay of 5 seconds between each request.
+ * See [maxRetries] and [minDelaySeconds]
+ */
+class RateLimitExceededEvent internal constructor(
```

Comment:
You'll need to run `./gradlew apiDump` before lint will pass.

(PS, I normally do `./gradlew apiDump testDebugUnitTest spotlessApply` before committing).